src/Communicants/Communicant.cpp

```diff
@@ -235,6 +234,29 @@ void Communicant::commSend(const ZZ_p& num) {
     commSend(ustring(toSend, MOD_SIZE), MOD_SIZE);
 }
 
+void Communicant::commSend(const vec_ZZ& vec) {
+    Logger::gLog(Logger::COMM, "... attempting to send: vec_ZZ " + toStr(vec));
+
+    ZZ max_elem = ZZ(0);
+    for (long i = 0; i < vec.length(); ++i) {
+        if (vec[i] > max_elem) max_elem = vec[i];
+    }
+    // plus 2 to avoid collision
+    ZZ base = max_elem + 2;
+
+    // send base first
+    commSend(base);
+
+    // pack vec_zz into a big zz
+    ZZ result = ZZ(0);
+    for (long i = vec.length() - 1; i >= 0; --i) {
+        result = result * base + vec[i] + 1; // add 1 to avoid ambiguity with zero
```

Comment:
I don't think this +1 trick works ... what if vec[i]==-1?  You probably have to translate the entire range min_elem to max_elem into an integer in the range 0...(max_elem-min_elem).